### ISSUE TITLE
Enhancement: Command-line startup of profile

### ISSUE THREAD

**Issue by robert914** [3noix/UJPS#18]
I don't think the UJPS.exe supports this currently, but it would be nice if we could specify from the command prompt the executable and profile to load up and start running. This would be useful for calling UJPS and starting a profile from things like VoiceAttack or AHK. I like to setup VoiceAttack commands for starting up games sometimes (e.g. startup UJPS with profile, startup supporting applications, startup game launcher), so it would be nice if we could do this.

Maybe, something like this:
```
UJPS.exe --profile C:\Path\To\Profiles\ProfileName.pro --run --minimized
```

Options:
`--profile` = profile to load into UJPS
`--run` = start profile running after loading (leaving this off just causes UJPS to load the profile only)
`--minimized` = start UJPS minimized to the tray

That would allow flexibility in automating the startup of UJPS if you were calling it from other programs (e.g. VoiceAttack, AHK, etc.).

**Comment by 3noix** [comment]
Very good idea! I put it on my list!

**Comment by 3noix** [comment]
I released a first version with a way to run UJPS using command line in this last commit.

I had to do a separate executable because for GUI applications (according to what is specified in the .pro file, i.e. even in the code we do not ask to create and show a window) for 2 reasons:

- I have seen that std::cout and std::cerr are not directed to the console, this means you should have any message about the profile status or errors (as in the UJPS main window).

- the other issue was that when killing the program or close the console, cmd was killed but not UJPS.

**Comment by 3noix** [comment]
And I have not found a clean way to end the profile execution without killing UJPS. And killing it means that (Abstract)Profile::stop function is not called. I for example use it to turn off the leds of Warthog throttle.

**Comment by 3noix** [comment]
I provided a .bat example in the "Profiles" subdirectory

**Comment by 3noix** [comment]
I updated it with the possibility to add the time step (in milliseconds) as a second optional argument. If not provided, 15 ms is used.

**Comment by robert914** [comment]
Sorry for the late response on this. I was out of town this past weekend. I finally got a change to try this, and it does work fine running from a terminal and not opening any GUI. However, I was somewhat expecting that the UJPS interface would still open with this method and basically just skip all the Browse / Run /etc steps and be running with the interface opened (unless you specify some option to turn that off).

The reason I was expecting it, is with the current UJPS_console.exe, my profile that has the UI dialog that appears, won't run now. It crashes stating it cannot create a QWidget since it's not a QApplication.

Is there anyway you can set this up so I can do the above  but still have the interface appear? I can see benefits to the current way you have it implemented, so I don't know that I would eliminate what you've done.

**Comment by robert914** [comment]
I suppose if there is some way I can detect if the Qt environment is available, I can switch my profile code to prompt in the terminal instead of using a Dialog. I'm not exactly sure how I can tell if the Qt environment is usable without it being there?

**Comment by robert914** [comment]
Ok, I found a way to detect if it's running console mode by checking this:
```
QCoreApplication::arguments().at(0)
```
I can check if it's the "UJPS_console.exe" binary, or the "UJPS.exe" binary. Then I can change my code to prompt via console or dialog based on that.

I don't think you need to change anything. Thanks for getting this working!

**Comment by 3noix** [comment]
Yes, no problem, I will add the possibility to specify the same 2 arguments (profile - mandatory - and time step - optional) for the GUI version, just to pre-fill the profile path and time step. I shouldn't have closed the issue before that!

**Comment by robert914** [comment]
Can it also start the profile running?

**Comment by 3noix** [comment]
Yes, it is possible. I will add an option for that. And as the loading of the project can take a few seconds, if this option is selected, I will display a splashscreen (to avoid to see the main window frozen), before showing the main window.

**Comment by robert914** [comment]
Great! Sounds good.

**Comment by robert914** [comment]
I think I found a minor bug. If I do one of the following:
```
UJPS.exe C:\Users\robert\Profiles\StarCitizen\StarCitizen.pro -play
```
or
```
cd C:\Users\robert\Profiles
UJPS.exe StarCitizen\StarCitizen.pro -play
```
or
```
cd C:\Users\robert\Profiles\StarCitizen
UJPS.exe .\StarCitizen.pro -play
```
They all load and run the profile just fine. However, try to run the profile from within the profile directory like this without any ".\" or full path:
```
cd C:\Users\robert\Profiles\StarCitizen
UJPS.exe StarCitizen.pro -play
```
It fails and says: "No profile dll in directory /release" 
For some reason it doesn't see that the profile is in the current directory and properly resolve the location of the "release" directory.

**Comment by 3noix** [comment]
Corrected! Here is the detailed description of the commit:
- correct the issue with some relative profile paths
- move many messages from main window to "profile engine" -> all "gui non-related" messages are visible in the console for the console version. Before only a few were sent to the console

**Comment by robert914** [comment]
Cool! Works now. Thanks for this! I can see uses for both the GUI and console based exe. 👍
